x-pack/examples/ui_actions_enhanced_examples/public/drilldowns/canvas_to_dashboard_drilldown/canvas_to_dashboard_drilldown.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  DashboardEnhancedAbstractDashboardDrilldown as AbstractDashboardDrilldown,
+  DashboardEnhancedAbstractDashboardDrilldownConfig as Config,
+} from '../../../../../plugins/dashboard_enhanced/public';
+import {
+  SAMPLE_CANVAS_ELEMENT_CLICK_TRIGGER,
+  SampleCanvasElementClickContext,
+} from '../../triggers';
+import { KibanaURL } from '../../../../../../src/plugins/share/public';
+
+export const SAMPLE_CANVAS_TO_DASHBOARD_DRILLDOWN = 'SAMPLE_CANVAS_TO_DASHBOARD_DRILLDOWN';
+
+type Trigger = typeof SAMPLE_CANVAS_ELEMENT_CLICK_TRIGGER;
+type Context = SampleCanvasElementClickContext;
+
+export class SampleCanvasToDashboardDrilldown extends AbstractDashboardDrilldown<Trigger> {
```

Comment:
For me personally it is not easy to follow this pattern with abstract factory. I know we already introduced similar with `expore_underlying_data` and I didn't like it there earthier. 
I'd prefer just extracting common logic into functions and reusing them. For me it is easier to follow even though it leads to a bit more code.

In an intention of "one way of doing things", do you think it worth brining this up to wider team for discussion?

Reply:
The idea is simple, you extend that class and all you need to do is implement is the `getURL` function, and you have Dashboard drilldown working in your app.

Before arriving at this approach, I've tried a couple other options, including just creating a separate drilldown and reusing the React components of Dashboard drilldown and config functions, but that created a lot of messy imports, thus settled for this approach. With this approach it is only 15 lines of code to define a dashboard drilldown in your app.

But curious to see if you have in mind a nicer way to do it. If you have in mind making all the functions like `execute`, `CollectConfig`, `validateConfig`, `isCompatible`, `getDisplayName`, `getIconType` as standalone, which can be imported by multiple drilldowns, I would not like it over the current solution; as it is one of the options I tried and it was messy to track all the imports. Here you really need to import just one thing, i.e. `AbstractDashboardDrilldown` and extend it.